### PR TITLE
Bump Istio to 1.23.2

### PR DESCRIPTION
**Pull request by werdes72**
**Description**

Changes proposed in this pull request:

- Update Istio version to 1.23.2
- Remove [obsolete environment variables](https://github.com/istio/istio/blob/1d4f23f70b4d6bf0a71995808393b7523199cc7c/releasenotes/notes/51987.yaml#L8)
- Bump manager Go version to 1.23.1
- Update environment variables related to [compatibility mode](https://github.com/istio/istio/blob/1.23.2/manifests/helm-profiles/compatibility-version-1.22.yaml)

**Pre-Merge Checklist**

- [x] As a PR reviewer, verify code coverage and evaluate if it is acceptable.

**Related issues**
- https://github.com/kyma-project/istio/issues/998

### CODE REVIEW

File: api/v1alpha2/compatibility_mode_test.go [hunk 1]
```diff
@@ -188,11 +188,11 @@ var _ = Describe("Compatibility Mode", func() {
...
 			Expect(err).ShouldNot(HaveOccurred())
 
-			xdsAgent := getProxyMetadataField(out, "BOOTSTRAP_XDS_AGENT")
+			xdsAgent := getProxyMetadataField(out, "ENABLE_DEFERRED_CLUSTER_CREATION")
 			Expect(xdsAgent).ToNot(BeNil())
-			Expect(xdsAgent.GetStringValue()).To(Equal("true"))
+			Expect(xdsAgent.GetStringValue()).To(Equal("false"))
 
-			deltaXds := getProxyMetadataField(out, "ISTIO_DELTA_XDS")
+			deltaXds := getProxyMetadataField(out, "ENABLE_DELIMITED_STATS_TAG_REGEX")
```
Comment (kolodziejczak) [inline api/v1alpha2/compatibility_mode_test.go:195]: Variable name should also be changed here

File: api/v1alpha2/compatibility_mode_test.go [hunk 2]
```diff
@@ -188,11 +188,11 @@ var _ = Describe("Compatibility Mode", func() {
 			//then
 			Expect(err).ShouldNot(HaveOccurred())
 
-			xdsAgent := getProxyMetadataField(out, "BOOTSTRAP_XDS_AGENT")
+			xdsAgent := getProxyMetadataField(out, "ENABLE_DEFERRED_CLUSTER_CREATION")
```
Comment (kolodziejczak) [inline api/v1alpha2/compatibility_mode_test.go:191]: Variable name should also be changed here

File: go.mod [hunk 3]
```diff
@@ -24,25 +24,24 @@ require (
 	golang.org/x/time v0.6.0
 	google.golang.org/protobuf v1.34.2
 	gopkg.in/yaml.v3 v3.0.1
-	istio.io/api v1.22.3
-	istio.io/client-go v1.22.3
-	istio.io/istio v0.0.0-20240703124446-b880f7040530
+	istio.io/api v1.23.1-0.20240906150629-ba126bb830f0
+	istio.io/client-go v1.23.1-0.20240906150928-c84358ed0e43
+	istio.io/istio v0.0.0-20240917190016-da0b3f29e9f9
```
Comment (kolodziejczak) [inline go.mod:29]: I think it should be 1.23.2 here

File: internal/istiooperator/istio-operator-light.yaml
Comment (kolodziejczak) [inline internal/istiooperator/istio-operator-light.yaml:1]: Can you point me to the documentation/source for changes here?

File: api/v1alpha2/compatibility_mode_test.go
Comment (kolodziejczak) [inline api/v1alpha2/compatibility_mode_test.go:1]: I think there are more tests that should be adjusted.

**Review by kolodziejczak — CHANGES_REQUESTED** [review]
Please update also compatibility mode documentation with the envs change

File: internal/istiooperator/istio-operator-light.yaml
Comment (werdes72) [reply internal/istiooperator/istio-operator-light.yaml:1]: It's in the PR description: https://github.com/istio/istio/blob/1d4f23f70b4d6bf0a71995808393b7523199cc7c/releasenotes/notes/51987.yaml#L8

Comment (kolodziejczak) [reply internal/istiooperator/istio-operator-light.yaml:1]: thanks!

**Review by kolodziejczak — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**